### PR TITLE
feat(std/node): Added implementation of os.homedir()

### PR DESCRIPTION
**Pull request by arwtyxouymz**
I found the implementation of `os.homedir()` was lack.
This is related to https://github.com/denoland/deno/issues/3802

### CODE REVIEW

File: node/os.ts [hunk 1]
```diff
@@ -136,7 +136,15 @@ export function getPriority(pid = 0): number {
 
 /** Returns the string path of the current user's home directory. */
 export function homedir(): string | null {
-  notImplemented(SEE_GITHUB_ISSUE);
+  switch (Deno.build.os) {
+    case "windows":
+      return Deno.env.get("HOMEPATH") || null;
```
Comment (kt3k) [inline node/os.ts:141]: I think 'USERPROFILE' is more reliable env var for pointing the home directory in windows. See this discussion for example https://github.com/denoland/deno/pull/3462

Node.js documentation also says it uses 'USERPROFILE' https://nodejs.org/api/os.html#os_os_homedir

**Review by kt3k — COMMENTED** [review]
Thank you for your contribution. I left one comment.

**Comment by arwtyxouymz** [comment]
@kt3k Thank you for your review. I updated to use `USERPROFILE` instead of `HOMEPATH`.

**Review by kt3k — APPROVED** [review]
LGTM 👍

**Pull request merged**